bindinfo/cache.go

```diff
@@ -144,14 +151,45 @@ func (bindCacheUpdater *BindCacheUpdater) Update(fullLoad bool) (err error) {
 	}
 	err = bindCacheUpdater.loadDiff(sql, newBc)
 	if err != nil {
-		return errors.Trace(err)
+		return err
+	}
+
+	bindCacheUpdater.globalHandle.Store(newBc)
+	return nil
+}
+
+func (bindCacheUpdater *BindCacheUpdater) updateOneBind(originSQL, bindSQL, defaultDB string, createTs, updateTs types.Time, status, charset, collation string) error {
+	bindCacheUpdater.lock.Lock()
+	defer bindCacheUpdater.lock.Unlock()
+
+	record := &bindRecord{
+		OriginalSQL: originSQL,
+		BindSQL:     bindSQL,
+		Db:          defaultDB,
+		Status:      status,
+		CreateTime:  createTs,
+		UpdateTime:  updateTs,
+		Charset:     charset,
+		Collation:   collation,
 	}
 
+	bc := bindCacheUpdater.globalHandle.Get()
+	newBc := make(map[string][]*bindMeta, len(bc))
+	for hash, bindDataArr := range bc {
+		newBc[hash] = append(newBc[hash], bindDataArr...)
+	}
+	bc = newBc
+	var err error
+	err = bc.appendNode(record, bindCacheUpdater.parser)
```

Comment:
It is confusing that we update the bc but store the `newBc` in the following lines.